### PR TITLE
RoR Milestone 6 creating a private events rails app

### PR DESCRIPTION
**Pull request by Berabjesus**

### CODE REVIEW

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  scope :past, -> { where('date < now()').order('created_at DESc') }
+  scope :upcoming, -> { where('date > now()').order('created_at DESc') }
+
+  # def self.past
+  #   self.where("date < now()").order('created_at DESc')
+  # end
+
+  # def self.upcoming
+  #   self.where("date > now()").order('created_at DESc')
+  # end
```
Comment (kubilaycaglayan) [inline app/models/event.rb]: :bulb: Let's delete the commented out code here, please :pray:

File: app/views/events/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    <%= f.date_field :date, class: 'form-control datepicker w-25 mb-3'  %>
+  </div>
+
+  <%= f.submit 'Create Event', class: 'btn btn-success'%>
+  <div>
+    <% if @event.errors.any?%>
+    <div class="alert alert-danger">
+      <% @event.errors.full_messages.each do |error|%>
+      <p class="larger-text"><%= error%></p>
+      <%end%>
```
Comment (kubilaycaglayan) [inline app/views/events/new.html.erb]: :bulb: No application logic in the views, please. You can create helper methods to get the same benefit.

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
#### Hello Team 👋
#### Status: Changes Required ♻️

Okay, this project is really great.
It is working well. You even did styling which makes it prettier! 🎉⭐
Thanks for your time and effort! 😊👏
Let's cover a few more details to make your project even better, together ⏬

## README && Database

- [x] If you are making a customized database configuration, you have to tell about it in the readme file. In this case, when we follow the instructions in the readme file we are not able to make the application work. Because you are using a username and password for the PostgreSQL that only exists in your own computers. In fact, for this project, you don't even have to use PostgreSQL. FYI 😇 So please add proper instructions to make the project work on different computers :pray: 

- [x] Don't forget to add test instructions, please.

## UsersController
###### `show` action
💡 I've tested your application. And it is already working as it has been intended. But, we want to keep our code well organized. To do that, in views we put less responsibility. You have already created very beautiful methods with scopes to separate upcoming and previous events in the model side. There is no need to write that same code in the views again. Instead, we will go step by step. First, in the controller, we will call those methods(those are in the models) and store the information in class variables to use them solely on the views side.

- [x] lists events created by user, i.e. `@created_events =  current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- [x] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

## No Logic In The Views

- [x] There should be NO application logic in the views nor helpers (no `Post.new`, `Comment.create`, `current_user.friends.find_by(...)`...). Everything connected with data processing should be encapsulated in models and used in controllers.

💡 Check inline comments below, please 🙏
- [x] Also check application.html.erb too, please.

**Resources**
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

## Tests

💡 Thank you for creating tests, please note that it is explicitly stated that we have to use RSpec and Capybara for testing. So please, transform your tests into appropriately.

- [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
- [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

## That's it, team! Please request another code review after you are done making the changes. 😇

With all my respect. 🙏
> Kubilay Caglayan | Technical Support Engineer

File: app/models/event.rb (hunk 1 above)
Comment (Berabjesus) [reply app/models/event.rb]: Corrected

File: app/views/events/new.html.erb (hunk 2 above)
Comment (Berabjesus) [reply app/views/events/new.html.erb]: Corrected

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️

I've tested your project and everything works as expected.

### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**